lib/urn.js

```diff
@@ -44,7 +44,77 @@ module.exports = {
       if (!urn.match(allowedProtocolsRegExp) || !urn.match(/^([^:]+:)+[^:]+$/)) {
         return false;
       }
+
+      // Parse ID portion of URN
+      var cat = urn.split(':').slice(0, -1)
+      var id = urn.split(':').slice(-1)[0];
+
+      // Split identity by each byte
+      var charArr = id.match(/.{1,2}/g);
+
+      // Match category of URN
+      switch (cat) {
+        case 'pbk:ec:secp256r1':
+          // Check for compressed/uncompressed ECDSA public keys
+          if (charArr[0] === '02' || charArr[0] === '03') {
+            // Check length of compressed public key
+            if (charArr.length !== 25) {
```

Comment:
Is this suppose to be 33? Since compressed key is just x (32 bytes) + 1 byte prefix.